### ISSUE TITLE
MultiheadAttention for the case when query size and key size are different

### ISSUE THREAD

**Issue by ehgh** [pytorch/pytorch#75704]
### 🐛 Describe the bug

MultiheadAttention class fails for the case when query and key size are different and they are of type numpy int64.

```
import torch.nn as nn
import numpy as np
```

This line of code produces an error when embed_dim is different than kdim and they are of type np.int64
```
att = nn.MultiheadAttention(np.int64(5), 1, kdim=np.int64(2),vdim=np.int64(2))
```

Root cause is line below. ```self._qkv_same_embed_dim``` is of type ```<class 'numpy.bool_'>``` and while its value is ```False``` the if statement check below fails:
```
if self._qkv_same_embed_dim is False:
```
Solutions: replace line above with:
```
if not self._qkv_same_embed_dim:
```
Reason it is failing:
The original if statement first checks if ```self._qkv_same_embed_dim``` is of type ```bool```  and then checks its value, which returns False when the variable is a numpy boolean.

Pytorch version: '1.9.0+cu102'
Python version: Python 3.6.9
[GCC 8.4.0] on linux

Error log:
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-3-b68ee676a11c> in <module>
      1 import torch.nn as nn
      2 import numpy as np
----> 3 att = nn.MultiheadAttention(np.int64(5), 1, kdim=np.int64(2),vdim=np.int64(2))

~/.local/lib/python3.6/site-packages/torch/nn/modules/activation.py in __init__(self, embed_dim, num_heads, dropout, bias, add_bias_kv, add_zero_attn, kdim, vdim, batch_first, device, dtype)
    947         self.add_zero_attn = add_zero_attn
    948 
--> 949         self._reset_parameters()
    950 
    951     def _reset_parameters(self):

~/.local/lib/python3.6/site-packages/torch/nn/modules/activation.py in _reset_parameters(self)
    953             xavier_uniform_(self.in_proj_weight)
    954         else:
--> 955             xavier_uniform_(self.q_proj_weight)
    956             xavier_uniform_(self.k_proj_weight)
    957             xavier_uniform_(self.v_proj_weight)

~/.local/lib/python3.6/site-packages/torch/nn/init.py in xavier_uniform_(tensor, gain)
    315         >>> nn.init.xavier_uniform_(w, gain=nn.init.calculate_gain('relu'))
    316     """
--> 317     fan_in, fan_out = _calculate_fan_in_and_fan_out(tensor)
    318     std = gain * math.sqrt(2.0 / float(fan_in + fan_out))
    319     a = math.sqrt(3.0) * std  # Calculate uniform bounds from standard deviation

~/.local/lib/python3.6/site-packages/torch/nn/init.py in _calculate_fan_in_and_fan_out(tensor)
    277 
    278 def _calculate_fan_in_and_fan_out(tensor):
--> 279     dimensions = tensor.dim()
    280     if dimensions < 2:
    281         raise ValueError("Fan in and fan out can not be computed for tensor with fewer than 2 dimensions")

AttributeError: 'NoneType' object has no attribute 'dim'
```

### Versions

```
Collecting environment information...
PyTorch version: N/A
Is debug build: N/A
CUDA used to build PyTorch: N/A
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.6 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: Could not collect
CMake version: version 3.10.2
Libc version: glibc-2.15

Python version: 2.7.17 (default, Mar 18 2022, 13:21:42)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-5.4.0-105-generic-x86_64-with-Ubuntu-18.04-bionic
Is CUDA available: N/A
CUDA runtime version: Could not collect
GPU models and configuration: 
GPU 0: Quadro K2200
GPU 1: NVIDIA GeForce RTX 2080 Ti

Nvidia driver version: 470.103.01
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: N/A

Versions of relevant libraries:
[pip] Could not collect
[conda] Could not collect
```

cc @jbschlosser @bhosmer @cpuhrsch

**Comment by jbschlosser** [comment]
Hey @ehgh, thanks for the report! Seems like a simple enough fix that should probably be done regardless. We'd accept a PR fixing this if you'd like to contribute.

**Comment by vitrioil** [comment]
Since this was open for sometime. Fix in #77857.
